scripts/lint-linkcheck.mjs

```diff
@@ -181,12 +250,55 @@ class BrokenLinkChecker {
 	pathnameToHtmlFilePath (pathname) {
 		return path.join(this.buildOutputDir, pathname, 'index.html');
 	}
+
+	/**
+	 * Attempts to find a Markdown source file for the given `pathname`.
+	 * 
+	 * Example: Given a pathname of `/en/some-page` or `/en/some-page/`,
+	 * searches for the source file in the following locations
+	 * and returns the first matching path:
+	 * - `${this.pageSourceDir}/en/some-page.md`
+	 * - `${this.pageSourceDir}/en/some-page/index.md`
+	 * 
+	 * If no existing file is found, returns `undefined`.
+	 */
+	tryFindSourceFileForPathname (pathname) {
+		const possibleSourceFilePaths = [
+			path.join(this.pageSourceDir, pathname, '.') + '.md',
+			path.join(this.pageSourceDir, pathname, 'index.md'),
+		];
+		return possibleSourceFilePaths.find(possiblePath => fs.existsSync(possiblePath));
+	}
+
+	/**
+	 * Attempts to find the given link `href` inside `input` and returns its index on a match.
+	 * 
+	 * Prevents false positive partial matches (like an href of `/en/install` matching
+	 * an input containing `/en/install/auto`) by requiring the characters surrounding a match
+	 * not to be a part of URLs in Markdown.
+	 */
+	indexOfHref (input, href) {
+		let i = input.indexOf(href);
+		while (i !== -1) {
+			// Get the characters surrounding the current match (if any)
+			let charBefore = input.substring(i - 1, i);
+			let charAfter = input.substr(i + href.length, 1);
```

Comment:
`substr` is deprecated I believe. I'd suggest using `slice` for both of these but `substring` works too.

Reply:
Interesting, I didn't know about the deprecated status. I'm actually going to use the array-style accessors then because I'm only interested in one character max each, and this perfectly captures that:
```js
let charBefore = input[i - 1] || '';
let charAfter = input[i + href.length] || '';
```

Reply:
Oh yeah, good idea!